src/main/java/com/redhat/devtools/lsp4ij/features/semanticTokens/LSPSemanticTokenTypes.java

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.features.semanticTokens;
+
+import org.eclipse.lsp4j.SemanticTokenTypes;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Custom semantic token types.
+ */
+@ApiStatus.Internal
+public class LSPSemanticTokenTypes {
+
+    // TypeScript "member"
```

Comment:
This is obviously hard-coded for now or JavaScript/TypeScript simply wouldn't work very well. It's the same token type we discussed previously, and I agree that there needs to be a first-class way for a language server to register custom semantic token types and modifiers and how they should be interpreted by LSP4IJ. See #871 for details on a more complete implementation.

I'd like to propose that this hard-coded exception be allowed right now and, once all of this work is merged, I'll take care of #871, either as a derivative of what's proposed there or whatever other approach is determined to be used instead.

Reply:
Please create an issue to avoid doing that. Have you planned to work on that?

Reply:
Yes, it's already captured in #871 as noted above.